sound/soc/sof/ipc3.c

```diff
@@ -830,6 +777,60 @@ static int ipc3_fw_ready(struct snd_sof_dev *sdev, u32 cmd)
 	return ipc3_init_reply_data_buffer(sdev);
 }
 
+static int sof_ipc3_get_reply(struct snd_sof_dev *sdev)
```

Comment:
Do you genuinely need to move an entire function? Is there no way to e.g. declare the dependency just before the function?

Such as adding before sof_ipc3_get_reply a line like:

`static int ipc3_fw_ready(struct snd_sof_dev *sdev, u32 cmd);`

The definition can still be after this function, it doesn't matter to the compiler as long as it can see the declaration. That should also cause less code churn. Up to you if you think this is better than moving the function itself.

Reply:
Both approaches solve the same problem. If the extra commit doesn't bother anyone, might as well go with the current one.

Reply:
I think we should go with Paul's suggestion. Moving function around will pollute the history and make investigations harder.